meta/test.pm

```diff
@@ -530,6 +530,30 @@ sub CreateStatEnumTest
     WriteTest "}";
 }
 
+sub EnumValueTest
+{
+    #
+    # Purpose is to check if enums with the same value on master when backport to other version
+    #
+
+    DefineTestName "enum_value_test";
+
+    WriteTest "{";
+    WriteTest "    int val = 0;";
+    # This is a enum which is only available in v1.7 branch.
+    WriteTest "    val = SAI_SWITCH_ATTR_TUNNEL_LOOPBACK_PACKET_ACTION;";
+    WriteTest "    TEST_ASSERT_TRUE(val == 186, \"enum SAI_SWITCH_ATTR_TUNNEL_LOOPBACK_PACKET_ACTION is not with value 186.\");";
+
+    WriteTest "    val = SAI_SWITCH_ATTR_TUNNEL_OBJECTS_LIST;";
+    WriteTest "    TEST_ASSERT_TRUE(val == 190, \"enum SAI_SWITCH_ATTR_TUNNEL_OBJECTS_LIST is not with value 190.\");";
+    WriteTest "    val = SAI_TUNNEL_ATTR_ENCAP_QOS_TC_AND_COLOR_TO_DSCP_MAP;";
+    WriteTest "    TEST_ASSERT_TRUE(val == 25, \"enum SAI_TUNNEL_ATTR_ENCAP_QOS_TC_AND_COLOR_TO_DSCP_MAP is not with value 25.\");";
+    WriteTest "    val = SAI_OBJECT_TYPE_SWITCH_TUNNEL;";
+    WriteTest "    TEST_ASSERT_TRUE(val == 95, \"enum SAI_OBJECT_TYPE_SWITCH_TUNNEL is not with value 95.\");";
+
+    WriteTest "}";
+}
+
```

Comment:
i would move this to saisanitycheck.c
also, why those enums must have specific values ? why this is sudenny needed here?

Reply:
briefly, we try to match the enum value between master and v1.7 (for warm reboot).

examples:
0. SAI_SWITCH_ATTR_SUPPORTED_FAILOVER_MODE
The is the precondition, this value already been there before current change.
in order to specify the situation happening between 1.7 and master, let's check the SAI_SWITCH_ATTR_SUPPORTED_FAILOVER_MODE, between v1.7 and master.
**in v1.7 SAI_SWITCH_ATTR_SUPPORTED_FAILOVER_MODE ==185**
**in master SAI_SWITCH_ATTR_SUPPORTED_FAILOVER_MODE == 189**

then there is already a difference, but in my current, i just want to add more attributes from master to v1.7, but not want to touch the historical items. 

With those difference, i want to add more attributes, but keep them with the same value with master
1. SAI_SWITCH_ATTR_TUNNEL_LOOPBACK_PACKET_ACTION
 this is a v1.7 only attribute, it only available in v1.7, then it must have a value
**in v1.7 WAS SAI_SWITCH_ATTR_TUNNEL_LOOPBACK_PACKET_ACTION = 186**
**in master None of this enum**

2. SAI_SWITCH_ATTR_TUNNEL_OBJECTS_LIST
 this is the attribute i need to add into v1.7, cause it is from master, then i can set the value
**in v1.7 set SAI_SWITCH_ATTR_TUNNEL_OBJECTS_LIST= 190**
**in master SAI_SWITCH_ATTR_TUNNEL_OBJECTS_LIST= 190**

3. SAI_TUNNEL_ATTR_ENCAP_QOS_TC_AND_COLOR_TO_DSCP_MAP
Cause we don't need IPSec related attributes, i need to set it
**in v1.7 set SAI_TUNNEL_ATTR_ENCAP_QOS_TC_AND_COLOR_TO_DSCP_MAP= 25**
or it will be
**in v1.7 set SAI_TUNNEL_ATTR_ENCAP_QOS_TC_AND_COLOR_TO_DSCP_MAP= 23**

cause I set this value for SAI_TUNNEL_ATTR_ENCAP_QOS_TC_AND_COLOR_TO_DSCP_MAP
then i need the flag
**@flags skip enum value check**

then those are values that need to be set and keep them same with master.

Any suggestion are welcomed.

Thanks.

Reply:
i made syncd between v1.7 and v1.9 which was master at a time here: https://github.com/opencomputeproject/SAI/pull/1344, but it was hanging too long and got out of sync before merging, this was also discussed on SAI community meeting, that if this syncd will be made then it will fail to warm boot for some vendors from v1.7 to v1.7 synced since binary values on SAI will differ :(

there is no good way to do this sync, and you adding forced enum values will still hit some issues in the near future since some enums will start overlap and attributes will have different meaning

my suggestion here is to keep v1.7 branch as a temporary, and phase out this branch and move to v1.9 or later and forget about v1.7

Reply:
For the values mismatch between versions, how about if we use the master branch as anchor and check previous version with the master? can we use the saisanitycheck, or saimetadatatest for go through all the enums?

Reply:
we mentioned 1.7 to 1.9, did we make sure 1.5 to 1.7 can make the upgrade warm reboot?

Reply:
was that v1.5 to v1.7 or v1.4 to v1.7 ? i lost track of warm boot on lower versions, and as for tracking via master it's not possible via sanity check since that is compiled against 1 set of headers, and it can't have multiple headers compiled, i used checkheaders.pl to find miss matched values and used that to manually create PR #1344 and it's a paint to do this manually, but there is no other way, since between those versions there are some new enums introduced, some were removed and changed type, and entire whole api was remove SR6 and was replaced by segmented route, and this is causing issues when comparing those branches

Reply:
> it's a paint to do this manually

Cannot be more agree with that and thanks for the explicit details.

Reply:
removed this validation in EnumValueTest